eventsources/common/aws/aws.go

```diff
@@ -53,9 +54,20 @@ func GetAWSCredFromVolume(access *corev1.SecretKeySelector, secret *corev1.Secre
 	if err != nil {
 		return nil, errors.Wrap(err, "can not find secret key")
 	}
+
+	var token string
+	if sessionToken != nil {
+		token, err = common.GetSecretFromVolume(sessionToken)
+		if err != nil {
+			log := logging.NewArgoEventsLogger()
+			log.Warnf("Failed to fetch AWS session token from volume: %s", err)
```

Comment:
Do not print the error but let the returned error to be handled by the invoker.

I don't think this should be a warning, instead, since it's configured (not nil), then it should be honored.

Reply:
Done